aten/src/ATen/native/native_functions.yaml

```diff
@@ -6355,11 +6384,12 @@
     SparseCPU, SparseCUDA: copy_sparse_
   autogen: copy_sparse_to_sparse, copy_sparse_to_sparse.out
 
+# By adding the AutogradNestedTensor this makes this function CompositeImplicit-like for nested tensors
 - func: unbind.int(Tensor(a -> *) self, int dim=0) -> Tensor(a)[]
   variants: function, method
   dispatch:
     CompositeExplicitAutograd: unbind
-    NestedTensorCPU, NestedTensorCUDA: NestedTensor_unbind
+    AutogradNestedTensor, NestedTensorCPU, NestedTensorCUDA: NestedTensor_unbind
```

Comment:
This should be updated to use the `CompositeImplicitAutogradNestedTensor` introduced here: https://github.com/pytorch/pytorch/pull/82754

Reply:
I tried this and it failed because: 
```
        assert len(composites_in_dispatch) <= 1 or (
            len(composites_in_dispatch) == 2
            and DispatchKey.CompositeImplicitAutograd in composites_in_dispatch
            and DispatchKey.CompositeImplicitAutogradNestedTensor
            in composites_in_dispatch
        ),
```
However fixing this logic to allow for compositeexplicit and compositeAGNT did not work either so need to actually think about it

Reply:
@bdhirsh codegen is being mean. hopfully I can figure this out before you get back

Reply:
oh yup, that check will need to be loosened a bit since we'll need this to be valid:
```
dispatch:
  CompositeExplicitAutograd
  CompositeImplicitAutogradNestedTensor
```

Maybe you ran into some additional problems though